src/sigma/remint.cpp

```diff
@@ -0,0 +1,111 @@
+#include "remint.h"
+#include "zerocoin.h"
+#include "libzerocoin/Commitment.h"
+
+namespace sigma {
+
+CoinRemintToV3::CoinRemintToV3(unsigned mintVersion, unsigned denomination, unsigned groupId, Bignum serial, Bignum randomness,
+                    uint256 originalMintBlockHash, const std::vector<unsigned char> &ecdsaPrivateKey) {
+    coinRemintVersion = 1;
+    coinMintVersion = mintVersion;
+    coinDenomination = denomination;
+    coinGroupId = groupId;
+    coinSerial = serial;
+    coinRandomness = randomness;
+    coinPublicValue = CalculatePublicValue();
+
+    if (coinMintVersion > ZEROCOIN_TX_VERSION_1) {
```

Comment:
Maybe we don't allow remint of zerocoins <verison 1.5? This is not secure, someone may just change these transaction outputs and take the funds. We can think what we can do with those mints, if there are any left in the blockchain?

Reply:
It's not set in stone if we need zerocoin mint v1 remint. I just left it here in case we need it later